symphony-bdk-core/src/main/java/com/symphony/bdk/core/activity/ActivityContext.java

```diff
@@ -0,0 +1,32 @@
+package com.symphony.bdk.core.activity;
+
+import com.symphony.bdk.gen.api.model.V4Initiator;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.apiguardian.api.API;
+
+/**
+ * Base class for an activity context holder. Contains the mandatory context attributes:
+ * <ul>
+ *   <li>{@link ActivityContext#initiator} : user info that triggered the chat event</li>
+ *   <li>{@link ActivityContext#eventSource} : chat event source object</li>
+ * </ul>
+ */
+@Getter
+@RequiredArgsConstructor
+@API(status = API.Status.STABLE)
+public abstract class ActivityContext<E> {
+
+  /** The activity initiator  (i.e. the Symphony that triggered an event in the chat)  */
+  private final V4Initiator initiator;
+
+  /**
+   * The activity real-time event source. Possible classes:
+   * <ul>
+   *   <li>{@link com.symphony.bdk.gen.api.model.V4MessageSent}</li>
+   *   <li>{@link com.symphony.bdk.gen.api.model.V4SymphonyElementsAction}</li>
+   * </ul>
+   */
+  private final E eventSource;
```

Comment:
I used to read `EventSource` in different existing APIs, especially in the HTM5 one : https://developer.mozilla.org/en-US/docs/Web/API/EventSource. But I can agree to rename if to `sourceEvent` if you think it would be clearer for our API users

Reply:
I think `EventSource` means a source which sends `event`s to users (something similar to a stream or a connection I guess). As we can see in the doc, it has some methods like `EventSource#close`, `EventSource#onOpen` ..., while in our case, our variable is just a single event.